lib/libc/man/mach_init.3
.\"
.\" $Id: mach_init.3,v 1.2 90/01/09 15:56:30 bww Exp $
.\"
.\" HISTORY
.\" $Log:	mach_init.3,v $
.\" Revision 1.2  90/01/09  15:56:30  bww
.\" 	Canonicalized NAME section.
.\" 	[90/01/09  15:56:14  bww]
.\" 
.\" Revision 1.1  89/12/28  14:39:51  bww
.\" 	Mach Release 2.5
.\" 	[89/12/28  14:39:00  bww]
.\" 
.\" 19-Nov-86  Mary Thompson (mrt) at Carnegie-Mellon University
.\"	Created.
.\"
.TH MACH_INIT 3 11/19/86
.SH NAME
.nf
task_self_ \- the port to which a task sends its kernel requests
task_data_ \- a port on which a task may receive messages
task_notify_ \- the port on which a task receives notify messages from the kernel
name_server_port \- the service port for the network name server
environment_port \- the service port for the task's environment manager
service_port \- the service port for the service server
mach_init_ports \- set of server ports initially handed to a task
mach_init_ports_count \- number of mach_init_ports
vm_page_size \- size of the virtual memory pages in bytes
.SH SYNOPSIS
#include <mach_init.h>
.SH DESCRIPTION
.I mach_init.h
exports the standard system and server ports and
global variables to the user.

.I mach_init.c
is called before the main program is entered to intitalize
the exported values and to initialize any of the standard
server interfaces. The standard server interfaces are:
.I init_mach, init_msgn, init_netname
and
.I init_env_mgr.
The service server should only be called by other servers, so it 
is not initialized here for the general user.

.I mach_init 
also initialzes the value of
.I vm_page_size
which is the virtual memory page size in bytes. Kernel requests dealing
with virtual memory are always rounded up to virtual pages. Occasionally
a user needs to know how big this is.

.SH FILES
/usr/include/mach_init.h

.SH BUGS
.I task_data_
and 
.I task_notify_
are the same port at this time. The port is initially
unrestricted. In the future they will be two different
ports and 
.I task_notify
will be initially restricted. (i.e. will not be included
in the set of ports used by a receive with 
.I PORT_DEFAULT
set.)


X11/CHANGES
The following changes have been made to this directory since R3:

     o	the configuration files have moved from util/imake.includes/ to config/
     o	sources for imake are now in config/ instead of util/imake/.



usr/lib/libmach_sa/mach_init.c
/* 
 * Mach Operating System
 * Copyright (c) 1991,1990,1989,1988,1987 Carnegie Mellon University
 * All Rights Reserved.
 * 
 * Permission to use, copy, modify and distribute this software and its
 * documentation is hereby granted, provided that both the copyright
 * notice and this permission notice appear in all copies of the
 * software, derivative works or modified versions, and any portions
 * thereof, and that both notices appear in supporting documentation.
 * 
 * CARNEGIE MELLON ALLOWS FREE USE OF THIS SOFTWARE IN ITS "AS IS"
 * CONDITION.  CARNEGIE MELLON DISCLAIMS ANY LIABILITY OF ANY KIND FOR
 * ANY DAMAGES WHATSOEVER RESULTING FROM THE USE OF THIS SOFTWARE.
 * 
 * Carnegie Mellon requests users of this software to return to
 * 
 *  Software Distribution Coordinator  or  Software.Distribution@CS.CMU.EDU
 *  School of Computer Science
 *  Carnegie Mellon University
 *  Pittsburgh PA 15213-3890
 * 
 * any improvements or extensions that they make and grant Carnegie Mellon
 * the rights to redistribute these changes.
 */
/*
 * HISTORY
 * $Log:	mach_init.c,v $
 * Revision 2.4  91/05/14  17:53:00  mrt
 * 	Correcting copyright
 * 
 * Revision 2.3  91/02/14  14:17:35  mrt
 * 	Added new Mach copyright
 * 	[91/02/13  12:44:06  mrt]
 * 
 * Revision 2.2  90/06/02  15:12:28  rpd
 * 	Converted to new IPC.
 * 	[90/03/26  23:26:00  rpd]
 * 
 * Revision 2.1  89/08/03  17:05:42  rwd
 * Created.
 * 
 * 18-Jan-89  David Golub (dbg) at Carnegie-Mellon University
 *	Altered for stand-alone use:
 *	. Removed registered port list.
 *	. Removed task_data (obsolete), task_notify.
 *	. Removed Accent compatibility calls.
 *
 * 23-Nov-87  Mary Thompson (mrt) at Carnegie Mellon
 *	removed includes of <servers/msgn.h> and <servers/netname.h>
 *	as they are no longer used.
 *
 * 5-Oct-87   Mary Thompson (mrt) at Carnegie Mellon
 *	Added an extern void definition of mig_init to keep
 *	lint happy
 *
 * 30-Jul-87  Mary Thompson (mrt) at Carnegie Mellon
 *	Changed the intialization of the mig_reply_port to be
 *	a call to mig_init instead init_mach.
 *
 * 27-May-87  Richard Draves (rpd) at Carnegie-Mellon University
 *	Changed initialization of mach interface, because the
 *	new mig doesn't export an alloc_port_mach function.
 *
 * 15-May-87  Mary Thompson
 *	Removed include of sys/features.h and conditional
 *	compliations
 *
 *  4-May-87  Mary Thompson
 *	vm_deallocted the init_ports array so that brk might 
 *	have a chance to be correct.
 *
 * 24-Apr-87  Mary Thompson
 *	changed type of mach_init_ports_count to unsigned int
 *
 * 12-Nov-86	Mary Thompson
 *	Added initialization call to init_netname the new
 *	interface for the net name server.
 *
 *  7-Nov-86  Michael Young
 * 	Add "service_port" 
 *
 *  7-Sep-86  Michael Young (mwyoung) at Carnegie-Mellon University
 *	Added initialization for MACH_IPC, using mach_ports_lookup.
 *
 * 29-Aug-86  Michael Young (mwyoung) at Carnegie-Mellon University
 *	Added other interesting global ports.
 */

#define	MACH_INIT_SLOTS		1
#include "mach_init.h"
#include <mach/mach.h>

extern void mig_init();

mach_port_t	mach_task_self_ = MACH_PORT_NULL;

vm_size_t	vm_page_size;

int		mach_init()
{
	vm_statistics_data_t	vm_stat;
	kern_return_t		kr;

#undef	mach_task_self

	/*
	 *	Get the important ports into the cached values,
	 *	as required by "mach_init.h".
	 */
	 
	mach_task_self_ = mach_task_self();

	/*
	 *	Initialize the single mig reply port
	 */

	mig_init(0);

	/*
	 *	Cache some other valuable system constants
	 */

	(void) vm_statistics(mach_task_self_, &vm_stat);
	vm_page_size = vm_stat.pagesize;

	return(0);
}

int		(*mach_init_routine)() = mach_init;
